src/analysis/retail/shaman/enhancement/modules/dragonflight/Tier31.tsx

```diff
@@ -0,0 +1,116 @@
+import { TIERS } from 'game/TIERS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import Statistic from 'parser/ui/Statistic';
+import Abilities from 'parser/core/modules/Abilities';
+import { TALENTS_SHAMAN } from 'common/TALENTS';
+import Events, { CastEvent } from 'parser/core/Events';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import ItemSetLink from 'interface/ItemSetLink';
+import { SHAMAN_T31_ID } from 'common/ITEMS/dragonflight';
+import TalentSpellText from 'parser/ui/TalentSpellText';
+import { UptimeIcon } from 'interface/icons';
+import { formatNumber, formatPercentage } from 'common/format';
+
+const PRIMORDIAL_WAVE_COOLDOWN_PER_SUMMON = 7000;
+
+export default class extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+    spellUsable: SpellUsable,
+  };
+  protected abilities!: Abilities;
+  protected spellUsable!: SpellUsable;
+  protected primordialWaveTotalCooldownReduction: number = 0;
+  protected primordialWaveCooldownEffectiveReduction: number = 0;
+  protected primordialWaveCasts: number = 0;
+  protected twoSetDamageBonus: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active =
+      this.selectedCombatant.has4PieceByTier(TIERS.T31) &&
+      this.selectedCombatant.hasTalent(TALENTS_SHAMAN.PRIMORDIAL_WAVE_SPEC_TALENT);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS_SHAMAN.FERAL_SPIRIT_TALENT),
+      (e) => this.onFeralSpiritSummoned(e.timestamp, 2),
+    );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS_SHAMAN.PRIMORDIAL_WAVE_SPEC_TALENT),
+      this.onPrimordialWaveCast,
+    );
+  }
+
+  onPrimordialWaveCast(event: CastEvent) {
+    this.primordialWaveCasts += 1;
+    this.onFeralSpiritSummoned(event.timestamp + 1, 1);
+  }
+
+  onFeralSpiritSummoned(timestamp: number, count: number) {
+    this.primordialWaveTotalCooldownReduction += PRIMORDIAL_WAVE_COOLDOWN_PER_SUMMON * count;
+    this.primordialWaveCooldownEffectiveReduction += this.spellUsable.reduceCooldown(
+      TALENTS_SHAMAN.PRIMORDIAL_WAVE_SPEC_TALENT.id,
+      PRIMORDIAL_WAVE_COOLDOWN_PER_SUMMON * count,
+      timestamp,
+    );
+  }
+
+  get wasted() {
+    return (
+      this.primordialWaveTotalCooldownReduction - this.primordialWaveCooldownEffectiveReduction
+    );
+  }
+
+  get averageReduction() {
+    return this.primordialWaveCooldownEffectiveReduction / 1000 / this.primordialWaveCasts || 0;
```

Comment:
instead of relying on the `|| 0` to handle the "never cast pwave" cast, prob cleaner to do this:
```suggestion
    return this.primordialWaveCasts ? this.primordialWaveCooldownEffectiveReduction / 1000 / this.primordialWaveCasts : 0;
```

generally any comparison with NaN/Inf is sus and a frequent source of pain